mysql/tests/test_mysql.py

```diff
@@ -493,22 +493,59 @@ def test_statement_samples_collect(
 
 @pytest.mark.integration
 @pytest.mark.usefixtures('dd_environment')
-def test_statement_samples_rate_limit(aggregator, bob_conn, dbm_instance):
+def test_statement_samples_collect_rate_limit(aggregator, bob_conn, dbm_instance):
+    # test rate limiting of the main collection loop
+    collections_per_second = 10
+    dbm_instance['statement_samples']['collections_per_second'] = collections_per_second
     dbm_instance['statement_samples']['run_sync'] = False
-    dbm_instance['statement_samples']['collections_per_second'] = 0.5
-    query = "select name as nam from testdb.users where name = 'hello'"
     mysql_check = MySql(common.CHECK_NAME, {}, instances=[dbm_instance])
-    with closing(bob_conn.cursor()) as cursor:
-        for _ in range(5):
-            cursor.execute(query)
-            mysql_check.check(dbm_instance)
-            time.sleep(1)
-    events = aggregator.get_event_platform_events("dbm-samples")
-    matching = [e for e in events if e['db']['statement'] == query]
-    assert len(matching) == 1, "should have collected exactly one event due to sample rate limit"
-    metrics = aggregator.metrics("dd.mysql.collect_statement_samples.time")
-    assert 2 < len(metrics) < 6
+    mysql_check.check(dbm_instance)
+    sleep_time = 1
+    time.sleep(sleep_time)
+    max_collections = int(collections_per_second * sleep_time)
     mysql_check.cancel()
+    metrics = aggregator.metrics("dd.mysql.collect_statement_samples.time")
+    assert max_collections / 2 <= len(metrics) <= max_collections
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+def test_statement_samples_unique_plans_rate_limits(aggregator, bob_conn, dbm_instance):
+    # test unique sample ingestion rate limiting
+    cache_max_size = 30
+    dbm_instance['statement_samples']['run_sync'] = True
+    # samples_per_hour_per_query set very high so that within this test we will have at most one sample per
```

Comment:
This comment is wrong, right? If I understand correctly, `samples_per_hour_per_query` is being set to 1 (a _low_ value rather than a _high_ one) to cap the samples per query to 1.

Reply:
Yes good catch, fixed